--- NVIDIA-Linux-x86_64-340.108-no-compat32-patched/kernel/uvm/nvidia_uvm_linux.h	2026-08-13 14:08:03.853207748 +0200
+++ work/kernel-6.1-uvm-fix/uvm/nvidia_uvm_linux.h	2026-08-20 18:57:33.319113134 +0200
@@ -451,6 +451,19 @@
 }
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0)
+// Rel. commit "mm: introduce vma->vm_flags wrapper functions" (Suren Baghdasaryan, 26 Jan 2023)
+static inline void vm_flags_set(struct vm_area_struct *vma, vm_flags_t flags)
+{
+    vma->vm_flags |= flags;
+}
+
+static inline void vm_flags_clear(struct vm_area_struct *vma, vm_flags_t flags)
+{
+    vma->vm_flags &= ~flags;
+}
+#endif
+
 static inline void nv_vm_flags_set(struct vm_area_struct *vma, vm_flags_t flags)
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0)
