--- NVIDIA-Linux-x86-340.108/kernel/nv-mmap.c.orig	2026-02-28 20:46:00.425354037 +0200
+++ NVIDIA-Linux-x86-340.108/kernel/nv-mmap.c	2026-03-01 12:17:45.190551791 +0200
@@ -391,7 +391,7 @@
     return status;
 }
 
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
 static NvBool nv_vma_enter_locked(struct vm_area_struct *vma, NvBool detaching)
 {
     NvU32 tgt_refcnt = VMA_LOCK_OFFSET;
--- NVIDIA-Linux-x86-340.108/kernel/uvm/nvidia_uvm_linux.h.orig	2026-03-03 12:48:38.182580038 +0200
+++ NVIDIA-Linux-x86-340.108/kernel/uvm/nvidia_uvm_linux.h	2026-03-03 13:18:03.015549077 +0200
@@ -461,7 +461,7 @@
 #define NV_CAN_CALL_VMA_START_WRITE 1
 #endif
 
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
 /*
  * Commit 45ad9f5290dc updated vma_start_write() to call __vma_start_write().
  */
@@ -490,7 +490,7 @@
 
 static inline void nv_vm_flags_set(struct vm_area_struct *vma, vm_flags_t flags)
 {
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
     nv_vma_flags_set_word(vma, flags);
 #elif defined(NV_VM_FLAGS_SET_PRESENT)
     vm_flags_set(vma, flags);
@@ -501,7 +501,7 @@
 
 static inline void nv_vm_flags_clear(struct vm_area_struct *vma, vm_flags_t flags)
 {
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
     nv_vma_flags_clear_word(vma, flags);
 #elif defined(NV_VM_FLAGS_SET_PRESENT)
     vm_flags_clear(vma, flags);
--- NVIDIA-Linux-x86-340.108/kernel/nv-linux.h.orig	2026-03-03 12:48:38.435394201 +0200
+++ NVIDIA-Linux-x86-340.108/kernel/nv-linux.h	2026-03-03 13:07:25.520894568 +0200
@@ -2274,7 +2274,7 @@
 #define NV_CAN_CALL_VMA_START_WRITE 1
 #endif
 
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
 /*
  * Commit 45ad9f5290dc updated vma_start_write() to call __vma_start_write().
  */
@@ -2299,7 +2299,7 @@
     ACCESS_PRIVATE(vma, __vm_flags) &= ~flags;
 #endif
 }
-#endif // !NV_CAN_CALL_VMA_START_WRITE
+#endif // !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 0, 0)
 typedef unsigned long vm_flags_t;
@@ -2307,7 +2307,7 @@
 
 static inline void nv_vm_flags_set(struct vm_area_struct *vma, vm_flags_t flags)
 {
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
     nv_vma_flags_set_word(vma, flags);
 #elif defined(NV_VM_FLAGS_SET_PRESENT)
     vm_flags_set(vma, flags);
@@ -2318,7 +2318,7 @@
 
 static inline void nv_vm_flags_clear(struct vm_area_struct *vma, vm_flags_t flags)
 {
-#if !NV_CAN_CALL_VMA_START_WRITE
+#if !NV_CAN_CALL_VMA_START_WRITE && defined(CONFIG_PER_VMA_LOCK)
     nv_vma_flags_clear_word(vma, flags);
 #elif defined(NV_VM_FLAGS_SET_PRESENT)
     vm_flags_clear(vma, flags);
